lib/ansible/plugins/strategy/__init__.py

```diff
@@ -92,7 +93,9 @@ def results_thread_main(strategy):
             result = strategy._final_q.get()
             if isinstance(result, StrategySentinel):
                 break
-            else:
+            elif isinstance(result, CallbackSend):
+                strategy._tqm.send_callback(result.method_name, *result.args, **result.kwargs)
+            elif isinstance(result, TaskResult):
                 strategy._results_lock.acquire()
```

Comment:
I know you didn't change this, but shouldn't this be used as a context manager?

Reply:
Good point, I'll do that.

Reply:
Addressed.